nornir/core/inventory.py

```diff
@@ -196,42 +196,52 @@ def nornir(self, value):
             self._nornir = value
 
     @property
-    def host(self):
-        """String used to connect to the device. Either ``nornir_host`` or ``self.name``"""
-        return self.get("nornir_host", self.name)
+    def hostname(self):
+        """String used to connect to the device. Either ``hostname`` or ``self.name``"""
+        return self.get("hostname", self.name)
+
+    @property
+    def port(self):
+        """Either ``port`` or ``None``."""
+        return self.get("port")
 
     @property
     def username(self):
         """Either ``nornir_username`` or user running the script."""
-        return self.get("nornir_username", getpass.getuser())
+        return self.get("username", getpass.getuser())
 
     @property
     def password(self):
         """Either ``nornir_password`` or empty string."""
-        return self.get("nornir_password", "")
-
-    @property
-    def ssh_port(self):
-        """Either ``nornir_ssh_port`` or ``None``."""
-        return self.get("nornir_ssh_port")
+        return self.get("password", "")
 
     @property
-    def network_api_port(self):
-        """
-        For network equipment this is the port where the device's API is listening to.
-        Either ``nornir_network_api_port`` or ``None``.
-        """
-        return self.get("nornir_network_api_port")
-
-    @property
-    def os(self):
-        """OS the device is running. Defaults to ``nornir_os``."""
-        return self.get("nornir_os")
-
-    @property
-    def nos(self):
-        """Network OS the device is running. Defaults to ``nornir_nos``."""
-        return self.get("nornir_nos")
+    def platform(self):
+        """OS the device is running. Defaults to ``platform``."""
+        return self.get("platform")
+
+    def get_connection_parameters(
+        self, connection: Optional[str] = None
+    ) -> Dict[str, Any]:
+        if not connection:
```

Comment:
I was thinking about this and I have mixed feelings. My concern is that it is a little bit too much magic which can be not so intuitive, but I also understand the reasoning behind including this.
What I think could be less magic:
To form a dictionary with default Host attributes hostname, port, username, password, platform and then just update with connection_options. connection_options would always be provided in a format understandable by the backend library (for the plugins we are shipping with nornir, user-defined plugins can do anything). For example, if the backend library uses pass instead of password, if you want to override host password, in connection_options you must define pass. Then to pass one resulting dictionary to the plugin and let it form its own args. In the connect plugin it would be something like:
conn_params['pass'] = options.get('pass', options.get('password'))
This looks a little ugly, but it feels a little more explicit.

I don't know what is best here, 50/50.

Reply:
>   conn_params['pass'] = options.get('pass', options.get('password'))

The problems I see with that are:

1. It's going to bring inconsistent behaviors between drivers as everybody has to remember to implement exactly the same behavior.
2. It's going to confuse users as in some plugins "password" works and in others they have to use "pass".
3. That means I can't override the connection with code. I am talking about the possibility of calling `Host.open_connection("netmiko", my_user, my_pass,...)` directly. Even though I am being explicit the plugin might end up not using what I passed and revert to the inventory options.

> To form a dictionary with default Host attributes hostname, port, username, password, platform and then just update with connection_options

Isn't that exactly what this is doing? :D

Reply:
> 1.It's going to bring inconsistent behaviors between drivers as everybody has to remember to implement exactly the same behavior.

If you need to override connection options, you should always refer to the backend library docs. Users would need to manually install the library already.

> 2.It's going to confuse users as in some plugins "password" works and in others they have to use "pass".

Partially, override should be used only with the library docs.
If I use `ncclient` and currently my platform is set to `ios`, but I need to pass `cisco_iosxe`, I would want to put it in the `ncclient` format:

```
sw1:
    ...
    platform: ios
    ncclient_options:
        device_params:
            name: cisco_iosxe
```

If I override netmiko, I would put:
```
sw1:
    ...
    platform: ios
    netmiko_serial_options:
        device_type: terminal_server
```
Doesn't it make sense?

> 3. That means I can't override the connection with code. I am talking about the possibility of calling Host.open_connection("netmiko", my_user, my_pass,...) directly.

Override by providing a new `connection_options` dict formatted how the library expects it.

> Isn't that exactly what this is doing? :D
Almost...
See my alternative proposals below